lms/static/js/dashboard/track_events.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Track the different click events on dashboard.
+ */
+
+var edx = edx || {};
+
+(function ($, analytics) {
+    'use strict';
+
+    $(document).ready(function () {
+        var course_title_link = $(".course-title-link"),
+            course_image_link = $(".dashboard-course-image"),
+            enter_course_link = $(".enter-course-link"),
+            course_action_more = $(".wrapper-action-more"),
+            course_action_upgrade = $(".action-upgrade"),
+            find_courses_btn = $(".btn-find-courses"),
+            course_learn_verified = $(".learned-verified-track");
+
+        // Fire analytics events when the "course title link" button is clicked.
+        course_title_link.on("click", function (event) {
+            var courseKey = $(event.target).data("course-key");
+            analytics.track(
+                "edx.bi.dashboard.clicked_course_title_link",
+                {
+                    category: "dashboard",
+                    label: courseKey
+                }
+            );
+        });
+
+        // Fire analytics events when the "course image" is clicked.
+        course_image_link.on("click", function (event) {
+            var courseKey = $(event.target).parent().data("course-key");
+            analytics.track(
```

Comment:
Have you tested these events? Since you're using `analytics.track()`, some of these events may not be emitted before the browser loads a new page. You should be using `analytics.trackLink()` which inserts a small timeout to give the `track` call enough time to execute. For an example, see [here](https://github.com/edx/edx-platform/blob/master/lms/static/js/dashboard/legacy.js#L46).